stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/SingleIdpPromptPageResource.java

```diff
@@ -32,26 +36,35 @@
     private final IdpStubsRepository idpStubsRepository;
     private final ServiceListService serviceListService;
     private final SingleIdpConfiguration singleIdpConfiguration;
+    private final AllIdpsUserRepository allIdpsUserRepository;
 
     @Inject
     public SingleIdpPromptPageResource(
             IdpStubsRepository idpStubsRepository,
             ServiceListService serviceListService,
-            SingleIdpConfiguration singleIdpConfiguration) {
+            SingleIdpConfiguration singleIdpConfiguration,
+            AllIdpsUserRepository allIdpsUserRepository) {
         this.idpStubsRepository = idpStubsRepository;
         this.serviceListService = serviceListService;
         this.singleIdpConfiguration = singleIdpConfiguration;
+        this.allIdpsUserRepository = allIdpsUserRepository;
     }
 
     @GET
     public Response get(
             @PathParam(Urls.IDP_ID_PARAM) @NotNull String idpName,
-            @QueryParam(Urls.ERROR_MESSAGE_PARAM) Optional<ErrorMessageType> errorMessage) throws FeatureNotEnabledException {
+            @QueryParam(Urls.ERROR_MESSAGE_PARAM) Optional<ErrorMessageType> errorMessage,
+            @CookieParam(CookieNames.SINGLE_IDP_SESSION_COOKIE_NAME) String persistentId) throws FeatureNotEnabledException {
```

Comment:
don't store the persistentId in a cookie (even unencrypted) - suggest this should be a UUID or similar

Reply:
A session is now created when the user pre-registers and when the SSO resource is accessed, the user is attached to the new session and the old session deleted.